analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+In general there are a few cells that are only found in 2 of the tools, but it looks like for the most part Alevin-fry, Alevin-fry-sketch, and Kallisto are all identifying the same cells and a subset of the cellranger cells. Alevin is a little weird with some overlap but also some unique cells. 
+
+Let's look at the correlation of UMI's/cell across the different tools. We're only going to look at the cells that are in common. 
+
+```{r}
+# spread table for comparisons
+cell_qc_common_cor <- cell_qc_common %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(cell_id, sample),
+                     names_from = tool,
+                     values_from = sum) %>%
+  # drop rows with NA values to ease correlation calculations
+  tidyr::drop_na()
+```
+
+For simplicity, let's just look at correlation of each tool with cellranger
+```{r}
+cell_qc_common_cor %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+Overall, they all have really high correlations with cellranger, although it does look like Alevin-fry has the consistently highest correlation with kallisto consistently the lowest correlation. 
+
+```{r}
+ggplot(cell_qc_common_cor, aes(x = cellranger, y = alevin_fry)) +
+  geom_point(size = 0.5, alpha = 0.1) + 
+  facet_wrap(~ sample) + 
+  scale_x_log10() + 
+  scale_y_log10() + 
+  labs(x = "Cell Ranger UMI/cell", y = "Alevin Fry UMI/cell") + 
+  theme_classic()
+```
+
+What about looking at the correlation of genes/cell. I think this should be relatively the same as UMI/cell. 
+```{r}
+gene_qc_common_cor <- cell_qc_common %>%
+  # spread the mean expression stats to one column per caller
+  tidyr::pivot_wider(id_cols = c(cell_id, sample),
+                     names_from = tool,
+                     values_from = detected) %>%
+  # drop rows with NA values to ease correlation calculations
+  tidyr::drop_na()
+```
+
+```{r}
+gene_qc_common_cor %>% 
+  dplyr::group_by(sample) %>%
+  dplyr::summarize(
+    cr_al_cor = cor(cellranger, alevin, method = "spearman"),
+    cr_al_fry_cor =cor(cellranger, alevin_fry, method = "spearman"),
+    cr_al_fry_sketch_cor = cor(cellranger, alevin_fry_sketch, method = "spearman"),
+    cr_ka_cor = cor(cellranger, kallisto, method = "spearman")
+  )
+```
+There is some slight variation, but still quite hgih. Again, it looks like Alevin fry has the best, with Kallisto having the worst. 
```

Comment:
Notably, alevin-fry sketch is still outperforming kallisto. This was something I wanted to know about...